fastlane/actions/analyze_commits_canary.rb

```diff
@@ -0,0 +1,399 @@
+            verify_block: proc do |value|
+              UI.user_error!("No match for analyze_commits action given, pass using `match: 'expr'`") unless value && !value.empty?
+            end
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :releases,
+            description: "Map types of commit to release (major, minor, patch)",
+            default_value: { fix: "patch", feat: "minor" },
+            type: Hash
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :codepush_friendly,
+            description: "These types are consider as codepush friendly automatically",
+            default_value: ["chore", "test", "docs"],
+            type: Array,
+            optional: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :tag_version_match,
+            description: "To parse version number from tag name",
+            default_value: '\d+\.\d+\.\d+'
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :ignore_scopes,
+            description: "To ignore certain scopes when calculating releases",
+            default_value: [],
+            type: Array,
+            optional: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :canary,
+            description: "Canary deployment",
+            type: Boolean,
+            default_value: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :preid,
+            description: "Canary pre-release id",
+            type: String,
+            default_value: "alpha"
+          )
+        ]
+      end
+
+      def self.output
+        # Define the shared values you are going to provide
+        # Example
+        [
+          ['RELEASE_ANALYZED', 'True if commits were analyzed.'],
+          ['RELEASE_IS_NEXT_VERSION_HIGHER', 'True if next version is higher then last version'],
+          ['RELEASE_LAST_TAG_HASH', 'Hash of commit that is tagged as a last version'],
+          ['RELEASE_LAST_VERSION', 'Last version number - parsed from last tag.'],
+          ['RELEASE_NEXT_MAJOR_VERSION', 'Major number of the next version'],
+          ['RELEASE_NEXT_MINOR_VERSION', 'Minor number of the next version'],
+          ['RELEASE_NEXT_PATCH_VERSION', 'Patch number of the next version'],
+          ['RELEASE_NEXT_VERSION', 'Next version string in format (major.minor.patch)'],
+          ['RELEASE_NEXT_PRERELEASE_VERSION', 'Next prerelease version string in format (preid.pre_ver)'],
+          ['RELEASE_LAST_INCOMPATIBLE_CODEPUSH_VERSION', 'Last commit without codepush']
+        ]
+      end
+
+      def self.return_value
+        # If your method provides a return value, you can describe here what it does
+        "Returns true if the next version is higher then the last version"
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
+        ["xotahal"]
```

Comment:
Is this accurate?

Reply:
I forked this guy's plugin to use as a base and then added the canary/prerelease logic to it. I might separate it out into another plugin and just open source it myself.